duplex_http_call.go

```diff
@@ -73,15 +77,6 @@ func newDuplexHTTPCall(
 		request:           request,
 		responseReady:     make(chan struct{}),
 	}
-	if err != nil {
```

Comment:
This entire path isn't needed anymore since constructing the Request can't fail now.